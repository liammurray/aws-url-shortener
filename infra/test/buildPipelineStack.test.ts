import { expect as cdkExpect, haveResource } from '@aws-cdk/assert'
import { SynthUtils } from '@aws-cdk/assert'

import * as cdk from '@aws-cdk/core'
import BuildPipelineStack from '../src/BuildPipelineStack'

describe('Build pipeline stack', () => {
  let stack: BuildPipelineStack

  beforeAll(() => {
    const app = new cdk.App()

    stack = new BuildPipelineStack(app, 'test-stack', {
      devApiStackName: 'urls-api-dev',
    })
  })
  xtest('stack matches snapshot', () => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
  })
  test('API stack check', () => {
    cdkExpect(stack).to(
      haveResource('AWS::CodePipeline::Pipeline', {
        Name: 'UrlShortenerMaster',
      })
    )
  })
})
