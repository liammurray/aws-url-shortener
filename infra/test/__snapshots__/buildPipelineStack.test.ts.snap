// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build pipeline stack stack matches snapshot 1`] = `
Object {
  "Parameters": Object {
    "SsmParameterValuecicdUrlShortenergithubrepoC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/cicd/UrlShortener/github/repo",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuecicdcommongithubownerC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/cicd/common/github/owner",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuecicdcommonlambdaBucketC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/cicd/common/lambdaBucket",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuecicdcommonnotificationemailC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/cicd/common/notification/email",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "BuildPipeline2C8DF1D9": Object {
      "DependsOn": Array [
        "BuildPipelineRoleDefaultPolicy8F90FDB2",
        "BuildPipelineRoleBC8F2FE4",
      ],
      "Properties": Object {
        "ArtifactStore": Object {
          "EncryptionKey": Object {
            "Id": Object {
              "Fn::GetAtt": Array [
                "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
                "Arn",
              ],
            },
            "Type": "KMS",
          },
          "Location": Object {
            "Ref": "BuildPipelineArtifactsBucket1D12579B",
          },
          "Type": "S3",
        },
        "Name": "UrlShortenerMaster",
        "RestartExecutionOnUpdate": true,
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "BuildPipelineRoleBC8F2FE4",
            "Arn",
          ],
        },
        "Stages": Array [
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": Object {
                  "Branch": "master",
                  "OAuthToken": "{{resolve:secretsmanager:codebuild/github/token:SecretString:::}}",
                  "Owner": Object {
                    "Ref": "SsmParameterValuecicdcommongithubownerC96584B6F00A464EAD1953AFF4B05118Parameter",
                  },
                  "PollForSourceChanges": false,
                  "Repo": Object {
                    "Ref": "SsmParameterValuecicdUrlShortenergithubrepoC96584B6F00A464EAD1953AFF4B05118Parameter",
                  },
                },
                "Name": "Code",
                "Namespace": "Source_Code_NS",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "src",
                  },
                ],
                "RunOrder": 1,
              },
              Object {
                "ActionTypeId": Object {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": Object {
                  "Branch": "master",
                  "OAuthToken": "{{resolve:secretsmanager:codebuild/github/token:SecretString:::}}",
                  "Owner": Object {
                    "Ref": "SsmParameterValuecicdcommongithubownerC96584B6F00A464EAD1953AFF4B05118Parameter",
                  },
                  "PollForSourceChanges": false,
                  "Repo": "maketools",
                },
                "Name": "Tools",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "tools",
                  },
                ],
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": Object {
                  "EnvironmentVariables": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "[{\\"name\\":\\"COMMIT_ID\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"#{Source_Code_NS.CommitId}\\"},{\\"name\\":\\"COMMIT_BRANCH\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"#{Source_Code_NS.BranchName}\\"},{\\"name\\":\\"NPM_TOKEN_PARAM_KEY\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"/cicd/common/github/npmtoken\\"},{\\"name\\":\\"SAM_DEPLOY_TEMPLATE\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"cfn-deploy.yml\\"},{\\"name\\":\\"PACKAGE_OUTPUT_BUCKET\\",\\"type\\":\\"PLAINTEXT\\",\\"value\\":\\"",
                        Object {
                          "Ref": "SsmParameterValuecicdcommonlambdaBucketC96584B6F00A464EAD1953AFF4B05118Parameter",
                        },
                        "\\"}]",
                      ],
                    ],
                  },
                  "PrimarySource": "src",
                  "ProjectName": Object {
                    "Ref": "BuildPipelineBuild98CFDA2B",
                  },
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "src",
                  },
                  Object {
                    "Name": "tools",
                  },
                ],
                "Name": "Build",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "buildOutput",
                  },
                ],
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineBuildCodePipelineActionRole3BEC2A2D",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Build",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ActionMode": "REPLACE_ON_FAILURE",
                  "Capabilities": "CAPABILITY_AUTO_EXPAND,CAPABILITY_NAMED_IAM",
                  "ParameterOverrides": "{\\"ApiStage\\":\\"dev\\",\\"CommitInfo\\":\\"#{Source_Code_NS.BranchName}_#{Source_Code_NS.CommitId}\\"}",
                  "RoleArn": Object {
                    "Fn::GetAtt": Array [
                      "BuildPipelineDeployDevDeployDevStackRole5FE95492",
                      "Arn",
                    ],
                  },
                  "StackName": "UrlShortener-dev",
                  "TemplatePath": "buildOutput::cfn-deploy.yml",
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "buildOutput",
                  },
                ],
                "Name": "DeployDevStack",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineDeployDevDeployDevStackCodePipelineActionRole925B7D5F",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "DeployDev",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "BuildPipelineArtifactsBucket1D12579B": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "KMSMasterKeyID": Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "BuildPipelineArtifactsBucketEncryptionKey45F69BB7": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion",
                "kms:GenerateDataKey",
                "kms:TagResource",
                "kms:UntagResource",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineRoleBC8F2FE4",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineBuildRoleF57AC968",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineBuildRoleF57AC968",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineDeployDevDeployDevStackRole5FE95492",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineDeployDevDeployDevStackCodePipelineActionRole925B7D5F",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Delete",
    },
    "BuildPipelineArtifactsBucketEncryptionKeyAliasA58D147C": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AliasName": "alias/codepipeline-teststackbuildpipelined74073ad",
        "TargetKeyId": Object {
          "Fn::GetAtt": Array [
            "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
            "Arn",
          ],
        },
      },
      "Type": "AWS::KMS::Alias",
      "UpdateReplacePolicy": "Delete",
    },
    "BuildPipelineBuild98CFDA2B": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "CODEPIPELINE",
        },
        "Description": "Build, test and package to create deploy template",
        "EncryptionKey": Object {
          "Fn::GetAtt": Array [
            "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
            "Arn",
          ],
        },
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:4.0",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "Name": "UrlShortener",
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "BuildPipelineBuildRoleF57AC968",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": "buildspec-sam.yml",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "BuildPipelineBuildCodePipelineActionRole3BEC2A2D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineBuildCodePipelineActionRoleDefaultPolicy61D2509D": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineBuild98CFDA2B",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineBuildCodePipelineActionRoleDefaultPolicy61D2509D",
        "Roles": Array [
          Object {
            "Ref": "BuildPipelineBuildCodePipelineActionRole3BEC2A2D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildPipelineBuildRoleDefaultPolicy557D94A6": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "BuildPipelineBuild98CFDA2B",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "BuildPipelineBuild98CFDA2B",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "BuildPipelineBuild98CFDA2B",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "ssm:GetParameters",
                "ssm:GetParameter",
                "ssm:DescribeParamters",
                "ssm:GetParameterHistory",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:ssm:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":parameter/cicd/UrlShortener/*",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:ssm:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":parameter/cicd/common/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:PutObject",
                "s3:GetObject",
                "s3:CreateMultipartUpload",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "SsmParameterValuecicdcommonlambdaBucketC96584B6F00A464EAD1953AFF4B05118Parameter",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "SsmParameterValuecicdcommonlambdaBucketC96584B6F00A464EAD1953AFF4B05118Parameter",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineArtifactsBucket1D12579B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "BuildPipelineArtifactsBucket1D12579B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineBuildRoleDefaultPolicy557D94A6",
        "Roles": Array [
          Object {
            "Ref": "BuildPipelineBuildRoleF57AC968",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildPipelineBuildRoleF57AC968": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineDeployDevDeployDevStackCodePipelineActionRole925B7D5F": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineDeployDevDeployDevStackCodePipelineActionRoleDefaultPolicyB7639011": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "iam:PassRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineDeployDevDeployDevStackRole5FE95492",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineArtifactsBucket1D12579B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "BuildPipelineArtifactsBucket1D12579B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "cloudformation:CreateStack",
                "cloudformation:DeleteStack",
                "cloudformation:DescribeStack*",
                "cloudformation:GetStackPolicy",
                "cloudformation:GetTemplate*",
                "cloudformation:SetStackPolicy",
                "cloudformation:UpdateStack",
                "cloudformation:ValidateTemplate",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":cloudformation:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":stack/UrlShortener-dev/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineDeployDevDeployDevStackCodePipelineActionRoleDefaultPolicyB7639011",
        "Roles": Array [
          Object {
            "Ref": "BuildPipelineDeployDevDeployDevStackCodePipelineActionRole925B7D5F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildPipelineDeployDevDeployDevStackRole5FE95492": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cloudformation.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineDeployDevDeployDevStackRoleDefaultPolicy76FCA60B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineArtifactsBucket1D12579B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "BuildPipelineArtifactsBucket1D12579B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "*",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineDeployDevDeployDevStackRoleDefaultPolicy76FCA60B",
        "Roles": Array [
          Object {
            "Ref": "BuildPipelineDeployDevDeployDevStackRole5FE95492",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildPipelineRoleBC8F2FE4": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineRoleDefaultPolicy8F90FDB2": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "BuildPipelineArtifactsBucket1D12579B",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "BuildPipelineArtifactsBucket1D12579B",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineArtifactsBucketEncryptionKey45F69BB7",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineBuildCodePipelineActionRole3BEC2A2D",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "BuildPipelineDeployDevDeployDevStackCodePipelineActionRole925B7D5F",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineRoleDefaultPolicy8F90FDB2",
        "Roles": Array [
          Object {
            "Ref": "BuildPipelineRoleBC8F2FE4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildPipelineSourceCodeWebhookResource010CED8E": Object {
      "Properties": Object {
        "Authentication": "GITHUB_HMAC",
        "AuthenticationConfiguration": Object {
          "SecretToken": "{{resolve:secretsmanager:codebuild/github/token:SecretString:::}}",
        },
        "Filters": Array [
          Object {
            "JsonPath": "$.ref",
            "MatchEquals": "refs/heads/{Branch}",
          },
        ],
        "RegisterWithThirdParty": true,
        "TargetAction": "Code",
        "TargetPipeline": Object {
          "Ref": "BuildPipeline2C8DF1D9",
        },
        "TargetPipelineVersion": 1,
      },
      "Type": "AWS::CodePipeline::Webhook",
    },
  },
}
`;
